azext_iot/central/providers/v1/device_provider_v1.py

```diff
@@ -194,43 +194,75 @@ def get_device_registration_summary(self, central_dns_suffix=CENTRAL_ENDPOINT):
             central_dns_suffix=central_dns_suffix,
         )
 
-    def run_component_command(
+    def run_command(
         self,
         device_id: str,
         interface_id: str,
         command_name: str,
         payload: dict,
         central_dns_suffix=CENTRAL_ENDPOINT,
     ):
-        return central_services.device.run_component_command(
-            cmd=self._cmd,
-            app_id=self._app_id,
-            token=self._token,
+        if self._is_interface_id_component(
```

Comment:
Can you clarify the logic here? Or at least the path that gets us to this `run_command`?

If I provide an interface ID that doesn't exist in the `template.components` - this will attempt to run a device command ignoring the interface ID. The same goes for the `get_command_history` function - this will return all device history if the interface ID provided does not exist in the template components.

I'm not sure which provider (preview vs v1) is getting called where - but if `interface_id` is `None`, won't this be running unnecessary calls? At the very least it seems like we should add the truthy check 
`interface_id and` 
to this `if` statement

Reply:
good catch - fixed.